client/rest-high-level/src/main/java/org/elasticsearch/client/transform/StopTransformRequest.java

```diff
@@ -28,21 +28,23 @@
 
 public class StopTransformRequest implements Validatable {
 
+    public static final String WAIT_FOR_CHECKPOINT = "wait_for_checkpoint";
+
     private final String id;
     private Boolean waitForCompletion;
+    private Boolean waitForCheckpoint;
     private TimeValue timeout;
     private Boolean allowNoMatch;
 
     public StopTransformRequest(String id) {
-        this.id = id;
-        waitForCompletion = null;
-        timeout = null;
+        this(id, null, null, null);
     }
 
-    public StopTransformRequest(String id, Boolean waitForCompletion, TimeValue timeout) {
+    public StopTransformRequest(String id, Boolean waitForCompletion, TimeValue timeout, Boolean waitForCheckpoint) {
```

Comment:
I will have to think about being able to alert the user on if the _stop indeed caused the transform to stop at a checkpoint or not.

when `wait_for_completion` is spinning, it is checking on every cluster state update. Since we don't store this state information inside cluster state, there is nothing indicating if the value changed or not.

Reply:
@hendrikmuhs after digging around, I don't think this is possible. There are no hooks into how the task is cleared and we consider it "stopped" when it is deleted, consequently losing insight into how it was stopped.